File: A3/A3-012/A3-012.cpp
#include<iostream>
#include<vector>
using namespace std;

int N, S, ans = 0;
vector<int> toWho;
vector<bool> received;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);

    // Input student amount and start point
    cin >> N >> S;
    // Input message forward graph
    toWho.assign(N, 0);
    for(int i = 0; i < N; i++)
    {
        cin >> toWho[i];
        toWho[i]--;
    }

    // Count student amount who get a message
    received.assign(N, false);
    int curr = S - 1;
    while(true)
    {
        // The previous student didn't forward a message
        if(curr == -1) { break; }
        // The current student already got a message
        if(received[curr]) { break; }
        // The current student received a message
        received[curr] = true;
        ans++;  // Counting...
        // Go to the next student
        curr = toWho[curr];
    }

    // Output
    cout << ans << '\n';
    return 0;
}

File: A3/A3-012/A3-012.py
# Input student amount and start point
N, S = map(int, input().split())
# Input message forward graph
to_who = []
for i in range(N):
    to_who.append(int(input()) - 1)

# Count student amount who get a message
received = [False] * N
curr = S - 1
ans = 0
while(True):
    # The previous student didn't forward a message
    if(curr == -1):
        break
    # The current student already got a message
    if(received[curr]):
        break
    # The current student received a message
    received[curr] = True
    ans += 1    # Counting...
    # Go to the next student
    curr = to_who[curr]

# Output
print(ans)